src/Compilers/Core/Portable/InternalUtilities/ReadOnlyHashSet.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace Roslyn.Utilities
+{
+    internal struct ReadOnlyHashSet<T> : IReadOnlySet<T>
+    {
+        private HashSet<T> _set;
+
+        public static readonly ReadOnlyHashSet<T> Empty = new ReadOnlyHashSet<T>(new HashSet<T>());
```

Comment:
I think this should go in SpecializedCollections. It also seems a bit heavy weight for an empty set. I think we already have an Empty ISet<T>, would that work?